lib/matplotlib/backends/backend_agg.py

```diff
@@ -460,9 +460,14 @@ def draw(self):
         # acquire a lock on the shared font cache
         RendererAgg.lock.acquire()
 
+        toolbar = self.toolbar
         try:
+            if toolbar:
+                toolbar.set_cursor(cursors.WAIT)
```

Comment:
Sorry, missed the comments.
The reason to put this here is the same reason why canvas.toolbar exists on all canvases (possibly None) rather than only on interactive ones: because there is no InteractiveCanvas class in the hierarchy :-)  (Note that this is orthogonal with the toolmanager question)